code/clue.js

```diff
@@ -1,59 +1,347 @@
+  name: 'Guest House',
+  image: 'assets/guesthouse.jpg'
+}
+
+const patio = { 
+  name: 'Patio',
+  image: 'assets/patio.jpg'
+}
+
+// ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS 
 
 const suspects = [
   mrGreen,
-  mrsWhite
-  // ...  and the rest
+  mrMustard,
+  mrsPeacock,
+  mrPlum,
+  mrsScarlet,
+  mrsWhite 
 ]
 
-const weapons = []
+const weapons = [
+  rope,
+  knife,
+  candlestick,
+  dumbbell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol
+]
 
-const rooms = []
+const rooms = [
+  diningroom,
+  conservatory,
+  kitchen,
+  study,
+  library,
+  billiardsroom,
+  lounge,
+  ballroom,
+  hall,
+  spa,
+  livingroom,
+  observatory,
+  theater,
+  guesthouse,
+  patio
+]
 
 // THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
 const randomSelector = array => {
   return array[Math.floor(Math.random() * array.length)]
 }
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+// OBJECT THAT KEEPS THE MYSTERY.
+
+const mystery = {
+  killer: null,
+  weapon: null,
+  room: null
+};
+
+// RANDOM KILLER FUNCTION
+
+// FUNCTION TO DELAY PICK KILLER FUNCTION
+const delayPickKiller = () => {
+  document.getElementById('loaderKiller').style.display = 'block'
+  setTimeout(pickKiller, 1000)
+}
+document.getElementById('killerCard').addEventListener('click', delayPickKiller);
 
-// This function will be invoked when you click on the killer card.
+// PICK KILLER FUNCTION
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+  document.getElementById('loaderKiller').style.display = 'none'
+  // document.getElementById('killerCard').removeEventListener('click', delayPickKiller) // So the loader doesn't show in next click
+
+  mystery.killer = randomSelector(suspects) // This will randomly select a killer from the suspects. And add that to the mystery object.
+
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
+  // This will change what information about the killer is displayed
   document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('killerImage').src = mystery.killer.image
+  document.getElementById('killerOccupation').innerHTML = `${mystery.killer.occupation}`
+  document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('killerDescription').innerHTML = `${mystery.killer.description}`
+
+  // SHUFFLE FAVOURITE WEAPON
+  const shuffleFavouriteWeapon = () => {
```

Comment:
Good job, but in requirements we meant to shuffle favourite weapon of each suspect :)
Nonetheless, I consider this as done requirement